core/src/com/unciv/logic/civilization/ReligionManager.kt

```diff
@@ -87,19 +87,92 @@ class ReligionManager {
         if (storedFaith < faithForNextGreatProphet()) return false
         // In the base game, great prophets shouldn't generate anymore starting from the industrial era
         // This is difficult to implement in the current codebase, probably requires an additional variable in eras.json
-        // Also only if you either [have founded a religion] or [the max amount of religions (players/2 + 1) has not been reached].
-        // As this is yet to be implemented, this function does almost nothing
         return true
     }
+
+    private fun generateProphet() {
+        val prophetSpawnChange = (5f + storedFaith - faithForNextGreatProphet()) / 100f
+
+        if (Random(civInfo.gameInfo.turns).nextFloat() < prophetSpawnChange) {
+            val birthCity =
+                if (religionState == ReligionState.Pantheon) civInfo.getCapital()
+                else civInfo.cities.firstOrNull { it.id == religion!!.holyCityId }
+            val prophet = civInfo.addUnit("Great Prophet", birthCity)
+            if (prophet == null) return
+            prophet.religion = religion!!.name
+            prophet.abilityUsedCount["Religion Spread"] = 0
+            storedFaith -= faithForNextGreatProphet()
+        }
+    }
+
+    fun mayUseGreatProphetAtAll(prophet: MapUnit): Boolean {
+        if (religion == null) return false // First found a pantheon
+        if (religion!!.isMajorReligion()) return false // Already created a major religion
+        if (prophet.abilityUsedCount["Religion Spread"] != 0) return false // Already used its power for other things
+        if (civInfo.gameInfo.civilizations.count {
+                it.religionManager.religion != null && it.religionManager.religion!!.isMajorReligion()
+            }
+                >= civInfo.gameInfo.civilizations.count { it.isMajorCiv() } / 2 + 1
+            ) return false // Too bad, too many religions have already been founded.
+
+        return true
+    }
+
+    fun mayUseGreatProphetNow(prophet: MapUnit): Boolean {
+        if (!mayUseGreatProphetAtAll(prophet)) return false
+        if (!prophet.getTile().isCityCenter()) return false
+        return true
+    }
+
+    fun useGreatProphet(prophet: MapUnit) {
+        if (!mayUseGreatProphetNow(prophet)) return // How did you do this?
+        religionState = ReligionState.FoundingReligion
+        foundingCityId = prophet.getTile().getCity()!!.id
+    }
+
+    fun foundReligion(iconName: String, name: String, founderBelief: String, followerBelief: String) {
+        val newReligion = Religion(name, civInfo.gameInfo, civInfo.civName)
+        newReligion.iconName = iconName
+        if (religion != null) {
+            newReligion.pantheonBeliefs.addAll(religion!!.pantheonBeliefs)
+            newReligion.followerBeliefs.addAll(religion!!.followerBeliefs)
+            newReligion.founderBeliefs.addAll(religion!!.founderBeliefs)
```

Comment:
IIRC pantheon religions can't have any of these, no?

Reply:
True, but just in case. Worst case nothing is added, and if for some reason something weird happens (future modability?) which allows religions to be founded while you already have one, copying it over is probably the best way to do it anyway

Reply:
I disagree, code that's 'for the future, not in use' should not be added to the codebase.
It adds extra WTFs when human-parsing the code.